src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandlePropertyName.cs

```diff
@@ -142,11 +146,20 @@ public static partial class JsonSerializer
 #endif
                 if (jsonPropertyInfo.RuntimeClassInfo.CreateObject == null)
                 {
-                    ThrowHelper.ThrowNotSupportedException_SerializationNotSupported(jsonPropertyInfo.DeclaredPropertyType);
+                    // Special case for immutable dictionaries since we build up a dictionary and convert
+                    if (!jsonPropertyInfo.DeclaredPropertyType.IsImmutableDictionaryType())
+                    {
+                        ThrowHelper.ThrowNotSupportedException_SerializationNotSupported(jsonPropertyInfo.DeclaredPropertyType);
+                    }
+
+                    extensionData = Activator.CreateInstance(typeof(Dictionary<,>).MakeGenericType(genericArgs));
```

Comment:
We should be able to add a new abstract method on `JsonPropertyInfo`, and leverage the generic support on `JsonProperty<T>` to to create this instance. The call could look like this:

```
extensionData = jsonPropertyInfo.RuntimeClassInfo.ElementClassInfo.PropertyInfoForClassInfo.CreateDictionaryOfStringTValueInstance();
```